Code/Tools/AssetProcessor/Platform/Linux/native/FileWatcher/FileWatcher_linux.cpp

```diff
@@ -95,37 +164,103 @@ void FileWatcher::PlatformImplementation::AddWatchFolder(QString folder, bool re
     // Clean up the path before accepting it as a watch folder
     QString cleanPath = QDir::cleanPath(folder);
 
-    if (!TryToWatch(cleanPath))
+    if (source.IsExcluded(cleanPath))
+    {
+        DEBUG_FILEWATCHER("'%s' matches an exclusion rule.  Not watching.\n", cleanPath.toUtf8().constData());
+        return; // don't watch excluded paths and don't recurse into them.
+    }
+
+    if (!TryToWatch(cleanPath, nullptr, source))
     {
         return;
     }
+    
+    // Each watch costs linux a file handle from a limited (default 8k) set of handles.
+    // as such, we don't want to establish a watch on any excluded dirs.
+    // Its safer (and more efficient) to first establish the watches, then check for files.
+    QList<QString> dirsAdded;
+    dirsAdded.push_back(cleanPath);
 
-    // Add all the contents (files and directories) to watch and track them
     QDirIterator dirIter(
-        folder, QDir::NoDotAndDotDot | QDir::Dirs | QDir::Files,
+        folder, QDir::NoDotAndDotDot | QDir::Dirs,
         (recursive ? QDirIterator::Subdirectories : QDirIterator::NoIteratorFlags) | QDirIterator::FollowSymlinks);
 
     while (dirIter.hasNext())
     {
-        QString dirName = dirIter.next();
-        int theErrno;
-        if (!TryToWatch(dirName, &theErrno))
+        QString dirPath = dirIter.next();
+
+        if (source.IsExcluded(dirPath))
```

Comment:
Is dirPath a fullpath or just a path segment.
i.e Is it `/home/lawsonn/o3de/AutomatedTesting/Assets/textures` or is it `textures`?

If is the latter, won't files with the same name, but in different levels of the folder hierarchy be seen as already being watched?

Such as
`/home/lawsonn/o3de/AutomatedTesting/Assets/rendering/textures`
`/home/lawsonn/o3de/AutomatedTesting/Assets/audio/textures`

Reply:
its always a full path.